File: McvAngularTest2/Models/MyFormData.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace McvAngularTest2.Models
{
    public class MyFormData
    {
        public int? id { get; set; }
        public int locationId { get; set; }
        public string displayName { get; set; }

        public decimal? unitPrice { get; set; }
        public int startYear { get; set; }
        public string lastName { get; set; }
        public bool notifyViaMail { get; set; }

        public PersonNameVM extraPerson { get; set; }

        public string[] recipients { get; set; }

        public PersonNameVM[] contacts {get; set;}
    }
}

File: AngularForms/src/app/my-form/my-form-data.ts
import { PersonNameVM } from "./PersonNameVM";
import { AddressData } from "../address/addressData";

export interface MyFormData {
    id: number | null; // null for add
    locationId: number; // data to preserve
    displayName: string; // the name of the edited element, if in edit mode

    unitPrice: number | null;
    startYear: number;
    lastName: string;
    notifyViaMail: boolean;

    extraPerson: PersonNameVM;
    recipients: string[];
    contacts: PersonNameVM[];

    address: AddressData

}
